### PR TITLE
Update docs for `childOverlay` with custom component context

### PR DESCRIPTION
**Pull request by azlekov**
@arkivanov I have reorder a bit the location of the guidelines. Let me know if is not suitable. 

I found it more easy to look for Child Stack and Child Overlay usage with custom component context in respective pages rather than in the custom component page. But I left backlinks to be sure.

### CODE REVIEW

File: docs/navigation/overlay/component-context.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# How to initialize Child Overlay with custom ComponentContext
```
Comment (arkivanov) [inline docs/navigation/overlay/component-context.md]: Let's just write "Child Overlay with custom ComponentContext" here.

File: docs/navigation/overlay/component-context.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# How to initialize Child Overlay with custom ComponentContext 
+
+In order to pass this `AppComponentContext` to child overlay components, make an extension function on the `AppComponentContext` interface. This custom extension function will initialize the `Child Overlay` and provide child `AppComponentContext`.
```
Comment (arkivanov) [inline docs/navigation/overlay/component-context.md]: It seems that the underlying context of what are "custom component context" and `AppComponentContext` is missing here, if you just jump to this section without reading the `Custom ComponentContext` page. Maybe add something like the following?

> See [Custom ComponentContext](...) page for more information about creating custom `AppComponentContext`.

File: docs/navigation/overlay/component-context.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+```kotlin
+fun <C : Parcelable, T : Any> AppComponentContext.appChildOverlay(
```
Comment (arkivanov) [inline docs/navigation/overlay/component-context.md:8]: This sample looks very similar to `appChildStack`. Do you find dedicated samples actually helpful? What do you think if we just write something like the following after `appChildStack` samples?

> Creating `Child Overlay` with custom `ComponentContext` is very similar to `Child Stack`. Please refer to the samples above.

File: docs/navigation/stack/component-context.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# How to initialize Child Stack with custom ComponentContext
```
Comment (arkivanov) [inline docs/navigation/stack/component-context.md]: Let's just write "Child Stack with custom ComponentContext" here.

File: docs/navigation/stack/component-context.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# How to initialize Child Stack with custom ComponentContext 
+
+In order to pass this `AppComponentContext` to child components, make an extension function on the `AppComponentContext` interface. This custom extension function will initialize the `Child Stack` and provide child `AppComponentContext`.
```
Comment (arkivanov) [inline docs/navigation/stack/component-context.md]: Same concern with missing context here as above.

File: docs/component/custom-component-context.md [hunk 6]
```diff
@@ -22,64 +22,7 @@ class DefaultAppComponentContext(
 }
 ```
 
-## How to initialize Child Stack with custom ComponentContext
+## Custom ComponentContext with navigation
```
Comment (arkivanov) [inline docs/component/custom-component-context.md]: Maybe "Navigation with custom ComponentContext"?

**Review by arkivanov — CHANGES_REQUESTED** [review]

File: docs/navigation/overlay/component-context.md (hunk 1 above)
Comment (azlekov) [reply docs/navigation/overlay/component-context.md]: Thanks, done!

File: docs/navigation/overlay/component-context.md (hunk 2 above)
Comment (azlekov) [reply docs/navigation/overlay/component-context.md]: Nice catch I overseen this, thanks!

File: docs/navigation/overlay/component-context.md (hunk 3 above)
Comment (azlekov) [reply docs/navigation/overlay/component-context.md:8]: I did some changes here, please check. I will be a bit redundant but...

File: docs/navigation/stack/component-context.md (hunk 4 above)
Comment (azlekov) [reply docs/navigation/stack/component-context.md]: Done.

File: docs/navigation/stack/component-context.md (hunk 5 above)
Comment (azlekov) [reply docs/navigation/stack/component-context.md]: Done.

File: docs/component/custom-component-context.md (hunk 6 above)
Comment (azlekov) [reply docs/component/custom-component-context.md]: Done 👍

**Review by arkivanov — APPROVED** [review]

**Comment by arkivanov** [comment]
Thanks for your contribution!

**Pull request merged**